lightning/lib/bolt12/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+extern crate lightning;
+extern crate bech32;
+
+use std::convert::TryFrom;
+use std::ffi::{c_char, CStr};
+use lightning::offers::offer::Amount;
+use bech32::FromBase32;
+
+const BECH32_BOLT12_INVOICE_HRP: &str = "lni";
+
+#[no_mangle]
+pub unsafe extern "C" fn decode_offer(offer: *const c_char) -> *mut Offer {
+    let raw = CStr::from_ptr(offer);
+    let offer = match raw.to_str() {
+        Ok(res) => res,
+        Err(_) => return std::ptr::null_mut(),
```

Comment:
Wouldn't an error be nicer for showing "parsing failed because: <reason>"  and to avoid having to do null pointer checks?